lib/helpers.js

```diff
@@ -38,13 +40,23 @@ function getType(config) {
     // returns number[], boolean[], string[]
     return `${typeof firstElement}[]`;
   }
+  if (config instanceof RegExp) {
+    return 'RegExp';
+  }
   return typeof config;
 }
 
 function snakeCase(str) {
   return str.replace(/(\w)([A-Z]+)/g, (_, w, W) => `${w}_${W.toLowerCase()}`).replace(/\./g, '_');
 }
 
+function regex(str) {
+  const hasFlags = new RegExp('/.*/[dgimsuvy]*$').test(str);
+  const flags = hasFlags ? str.substring(str.lastIndexOf('/')).substring(1) : undefined;
+  const patternWithoutFlags = hasFlags ? str.substring(1, str.lastIndexOf('/')) : str;
+  return new RegExp(patternWithoutFlags, flags);
```

Comment:
🔧 instead of trying to get the flags from a regex I would get the flags from config. 
E.g pass the original config in castValue and use this (config.flags) instead of trying to overwrite it with a string